site/content/entry/accurate-audio-emulation.md

```diff
@@ -0,0 +1,199 @@
+Upon launching the game, everything seemed fine at first, but soon enough the familiar lack of background music kicked in, and Pokémon X started to behave exactly like it does in Citra.
+One short level up later, and the game froze, just like in Citra!
+We double checked the results by recreating this on different copies of Pokémon X and different 3dses, and it all went the exact same way.
+Audio issues is very likely the cause, but what should the fix be?
+
+{bleah i dunno what to put here}
+
+## The Long Road to LLE Audio Emulation 
+
+Fixing the audio issues now boils down to two potential solutions.
+Either take a good amount of time to research, reverse engineer, and recreate the audio code for Pokémon X/Y in HLE, or build out a program that can read the original binary audio code in Pokémon X/Y and emulate the actual audio chip, known as LLE.
+Each of them have there advantages and disadvantages: HLE means it'll be faster overall to get working, but also will likely take a lot more time and effort to fix bugs in other games with slightly different audio code; whereas LLE means that potentially every audio code in every game will *just work*, but also will take a lot longer to write, involve even more detailed technical research, and scariest of all, will probably end up running much slower.
+Weighing the risks and rewards, [wwylele](https://github.com/wwylele) ended up choosing to build out a full LLE audio emulator, what's now known as Teakra.
+The first commit on Teakra started in late January 2018, but design and research phase started before this.
+There is some documentation online about the audio chip, the TeakLite, which [wwylele](https://github.com/wwylele) referenced often throughout development.
+[GBATek](http://problemkaputt.de/gbatek.htm#dsixpertteakdsp) is loaded with valuable information about the processor, put together by the reverse engineering efforts of Martin Korth and many other contributors.
+
+Building an LLE audio emulator from scratch is an exciting, yet somewhat scary prospect.
+When working on HLE audio emulation, you really start to understand what the games are doing, but when writing an LLE audio emulator, you create a magical black box where binary data goes in, and binary data comes out.
+This is especially true for [wwylele](https://github.com/wwylele) because he had never worked with a DSP before this!
+It's daunting to know that you'll gain a deep understanding of the architecture and the instruction set of the audio chip, but you won't gain any detailed knowledge of how the audio code is functioning.
+
+{twitch clip of reading disassembled firmwares}
+
+## Wait, How Long Until This Starts Playing Sound?
+
+Several months after starting, the interpreter was coming along nicely, and it was time to start hooking things together.
```

Comment:
Will a casual reader know what 'interpreter' means in this context, or should you explain that out just slightly more.